### PR TITLE
Add xarray

### PR DESCRIPTION
**Pull request by weiji14**
### Description

Package for working with N-D labeled arrays and datasets in Python! Repo is at https://github.com/pydata/xarray

Crossref conda-forge recipe file: https://github.com/conda-forge/xarray-feedstock/blob/51aa230d50427cd3a93a8708fad8920e92ce3d62/recipe/meta.yaml

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

**Comment by hoodmane** [comment]
You need to require `typing-extensions` with a dash instead of an underscore.

File: packages/xarray/meta.yaml
Comment (weiji14) [inline packages/xarray/meta.yaml:13]: > You need to require `typing-extensions` with a dash instead of an underscore.

Thanks for the pointer @hoodmane. I'm actually not 100% sure if `typing-extensions` is needed at runtime. According to https://github.com/pydata/xarray/blob/9b4d0b29c319f4b68f89328b1bf558711f339504/setup.cfg#L77-L80, it seems like only `numpy`, `pandas` and `packaging` are required? It might just be that `xarray`'s conda-forge `recipe/meta.yaml` file hasn't been updated for a while.

**Comment by dopplershift** [comment]
General question: Does adding a pure Python package like this add some big benefit over just installing with micropip?

**Comment by weiji14** [comment]
> General question: Does adding a pure Python package like this add some big benefit over just installing with micropip?

Not sure, I was just trying to add a package that depends on `xarray` at https://github.com/pyodide/pyodide/pull/2525/commits/dde47d75eb19c72b5c8d59beb61c7068b70af3c5 and there was this error in the CI https://app.circleci.com/pipelines/github/pyodide/pyodide/4066/workflows/1a49c5b5-9111-425d-a055-c4ef255e2091/jobs/44836?invite=true#step-104-57:

```
ValueError: Directory /root/repo/packages/xarray does not contain meta.yaml
```

so I thought to add it in here. Is there an alternative way to add dependencies installable with `micropip`? I'm still very new to pyodide actually.

File: packages/xarray/meta.yaml
Comment (weiji14) [reply packages/xarray/meta.yaml:13]: So it seems to work without `typing-extensions`? `test_packages_common.py::import[firefox-xarray] PASSED                   [ 43%]`, see https://app.circleci.com/pipelines/github/pyodide/pyodide/4083/workflows/39debe58-bf65-4fe9-a1dc-4b96350cf193/jobs/45126?invite=true#step-102-157

**Comment by rth** [comment]
Yes, generally we are trying to avoid including pure Python packages in favor of using micropip https://github.com/pyodide/pyodide/issues/643 except when another package with C extensions needs it. Though even that would be resolved once #2132 is merged (unless it's a real build dependency)

The only advantage of including it is unvendoring of tests as a separate package. In the case of xarray, that's probably non-neglible (cf `build-packages-numpy-dependents` CI job, check_size step)
```
-rw-r--r-- 1 root root 507K May 10 18:13 xarray-2022.3.0-py3-none-any.whl
-rw-r--r-- 1 root root 2.0M May 10 18:13 xarray-tests.tar
```
it would be less after compression (the xarray wheel on Pypi is 870 kB), but still a not that small.

So for now we would probably need to package it indeed.

**Review by rth — APPROVED** [review]
Thanks LGTM. 
Could you add a changelog entry please?

File: packages/xarray/meta.yaml
Comment (rth) [reply packages/xarray/meta.yaml:13]: Yes, it's an optional dependency for py <3.10 https://github.com/pydata/xarray/blob/1941695ade6a3e4e46a07e6604f29bda5176de26/xarray/core/utils.py#L300 so it should be fine not to include it.

**Comment by hoodmane** [comment]
Could you add a `test_xarray` with a few basic tests?

**Review by hoodmane — APPROVED** [review]
Will merge if CI passes.

**Comment by hoodmane** [comment]
Thanks @weiji14!

**Comment by weiji14** [comment]
Cheers, thanks everyone!

**Pull request merged**
